frame/fnft/src/lib.rs

```diff
@@ -0,0 +1,450 @@
+			instance_id: FinancialNFTInstanceIdOf<T>,
+		},
+		FinancialNFTTransferred {
+			collection_id: FinancialNFTCollectionIdOf<T>,
+			instance_id: FinancialNFTInstanceIdOf<T>,
+			to: AccountIdOf<T>,
+		},
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		CollectionAlreadyExists,
+		InstanceAlreadyExists,
+		CollectionNotFound,
+		InstanceNotFound,
+		MustBeOwner,
+	}
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		#[allow(missing_docs)]
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		type MaxProperties: Get<u32>;
+
+		type FinancialNFTCollectionId: Parameter
+			+ Member
+			+ AssetIdLike
+			+ MaybeSerializeDeserialize
+			+ Ord
+			+ Into<u128>;
+
+		type FinancialNFTInstanceId: FullCodec
+			+ Debug
+			+ SafeAdd
+			+ MaxEncodedLen
+			+ Default
+			+ TypeInfo
+			+ Eq
+			+ PartialEq
+			+ Ord
+			+ Copy
+			+ Zero
+			+ One;
+
+		type ProxyType: Parameter + Member + Ord + PartialOrd + Default + MaxEncodedLen;
+
+		/// Used for setting the owning account of a fNFT as the delegate for the fNFT asset_account
+		type AccountProxy: AccountProxy<
+			AccountId = Self::AccountId,
+			ProxyType = Self::ProxyType,
+			BlockNumber = Self::BlockNumber,
+		>;
+
+		type ProxyTypeSelector: FNFTAccountProxyTypeSelector<Self::ProxyType>;
+
+		#[pallet::constant]
+		type PalletId: Get<PalletId>;
+	}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(_);
+
+	#[pallet::storage]
+	#[allow(clippy::disallowed_types)]
+	pub type FinancialNFTId<T: Config> = StorageMap<
+		_,
+		Blake2_128Concat,
+		FinancialNFTCollectionIdOf<T>,
+		FinancialNFTInstanceIdOf<T>,
+		ValueQuery,
+		NftIdOnEmpty<T>,
+	>;
+
+	#[pallet::type_value]
+	pub fn NftIdOnEmpty<T: Config>() -> FinancialNFTInstanceIdOf<T> {
+		Zero::zero()
+	}
+
+	#[pallet::storage]
+	#[pallet::getter(fn instance)]
+	pub type Instance<T: Config> = StorageMap<
+		_,
+		Blake2_128Concat,
+		(FinancialNFTCollectionIdOf<T>, FinancialNFTInstanceIdOf<T>),
+		(AccountIdOf<T>, BTreeMap<Vec<u8>, Vec<u8>>),
+		OptionQuery,
+	>;
+
+	/// Map of NFT collections to all of the instances of that collection.
+	#[pallet::storage]
+	#[pallet::getter(fn collection_instances)]
+	pub type CollectionInstances<T: Config> = StorageMap<
+		_,
+		Blake2_128Concat,
+		FinancialNFTCollectionIdOf<T>,
+		BTreeSet<FinancialNFTInstanceIdOf<T>>,
+		OptionQuery,
+	>;
+
+	/// All the NFTs owned by an account.
+	#[pallet::storage]
+	#[pallet::getter(fn owner_instances)]
+	pub type OwnerInstances<T: Config> = StorageMap<
+		_,
+		Blake2_128Concat,
+		AccountIdOf<T>,
+		BTreeSet<(FinancialNFTCollectionIdOf<T>, FinancialNFTInstanceIdOf<T>)>,
+		OptionQuery,
+	>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn collection)]
+	pub type Collection<T: Config> = StorageMap<
+		_,
+		Blake2_128Concat,
+		FinancialNFTCollectionIdOf<T>,
+		// (who, admin, data)
+		(AccountIdOf<T>, AccountIdOf<T>, BTreeMap<Vec<u8>, Vec<u8>>),
```

Comment:
Consider a struct instead of a tuple here

Reply:
As it's forked code, we can do these type of refactorings later :) , just want to keep this PR just for overall structure adjustments